### PR TITLE
Clarify bytes vs. text in TreeTag FBO Py3k.

### PR DESCRIPTION
**Pull request by tseaver**
`/manage_menu` was blowing up under Py3k with:

```python
Traceback (most recent call last):
  File "/tmp/z2py36/lib/python3.6/site-packages/waitress/channel.py", line 338, in service
    task.service()
  File "/tmp/z2py36/lib/python3.6/site-packages/waitress/task.py", line 169, in service
    self.execute()
  File "/tmp/z2py36/lib/python3.6/site-packages/waitress/task.py", line 399, in execute
    app_iter = self.channel.server.application(env, start_response)
  File "/tmp/z2py36/lib/python3.6/site-packages/ZPublisher/httpexceptions.py", line 33, in __call__
    return self.application(environ, start_response)
  File "/tmp/z2py36/lib/python3.6/site-packages/ZPublisher/WSGIPublisher.py", line 238, in publish_module
    request, response, module_info, _publish=_publish)
  File "/tmp/z2py36/lib/python3.6/site-packages/ZPublisher/WSGIPublisher.py", line 211, in _publish_response
    reraise(exc.__class__, exc, sys.exc_info()[2])
  File "/tmp/z2py36/lib/python3.6/site-packages/six.py", line 686, in reraise
    raise value
  File "/tmp/z2py36/lib/python3.6/site-packages/ZPublisher/WSGIPublisher.py", line 172, in _publish_response
    response = _publish(request, module_info)
  File "/tmp/z2py36/lib/python3.6/site-packages/ZPublisher/WSGIPublisher.py", line 162, in publish
    bind=1)
  File "/tmp/z2py36/lib/python3.6/site-packages/ZPublisher/mapply.py", line 85, in mapply
    return debug(object, args, context)
  File "/tmp/z2py36/lib/python3.6/site-packages/ZPublisher/WSGIPublisher.py", line 57, in call_object
    return obj(*args)
  File "/tmp/z2py36/lib/python3.6/site-packages/Shared/DC/Scripts/Bindings.py", line 323, in __call__
    return self._bindAndExec(args, kw, None)
  File "/tmp/z2py36/lib/python3.6/site-packages/Shared/DC/Scripts/Bindings.py", line 360, in _bindAndExec
    return self._exec(bound_data, args, kw)
  File "/tmp/z2py36/lib/python3.6/site-packages/App/special_dtml.py", line 203, in _exec
    result = render_blocks(self._v_blocks, ns)
  File "/home/tseaver/projects/Zope/Z2/DocumentTemplate/src/DocumentTemplate/_DocumentTemplate.py", line 147, in render_blocks
    render_blocks_(blocks, rendered, md)
  File "/home/tseaver/projects/Zope/Z2/DocumentTemplate/src/DocumentTemplate/_DocumentTemplate.py", line 248, in render_blocks_
    block = block(md)
  File "/home/tseaver/projects/Zope/Z2/DocumentTemplate/src/TreeDisplay/TreeTag.py", line 115, in render
    return tpRender(v, md, self.section, self.args)
  File "/home/tseaver/projects/Zope/Z2/DocumentTemplate/src/TreeDisplay/TreeTag.py", line 247, in tpRender
    section, md, treeData, level, args)
  File "/home/tseaver/projects/Zope/Z2/DocumentTemplate/src/TreeDisplay/TreeTag.py", line 527, in tpRenderTABLE
    colspan, section, md, treeData, level, args)
  File "/home/tseaver/projects/Zope/Z2/DocumentTemplate/src/TreeDisplay/TreeTag.py", line 386, in tpRenderTABLE
    l = s.find('=')
TypeError: a bytes-like object is required, not 'str'
```

### CODE REVIEW

File: src/TreeDisplay/TreeTag.py [hunk 1]
```diff
@@ -376,18 +376,7 @@ def tpRenderTABLE(self, id, root_url, url, state, substate, diff, data,
...
-            # Mostly inline encode_seq for speed
-            s = compress(json.dumps(diff))
-            if len(s) > 57:
-                s = encode_str(s)
-            else:
-                s = b2a_base64(s)[:-1]
-                l = s.find('=')
-                if l >= 0:
-                    s = s[:l]
-            s = s.translate(tplus)
```
Comment (hannosch) [inline src/TreeDisplay/TreeTag.py]: Does the translate call need to stay?

Comment (tseaver) [inline src/TreeDisplay/TreeTag.py]: `encode_str` [already takes care of it](https://github.com/zopefoundation/DocumentTemplate/pull/12/files/914fcd8fbc2e1a0188922cb70727bb9ac488480a#diff-75123b3701bcec58a795ebe5b2baf2f6R635).  The old code should've only been doing that in the "inlined" branch (which was an insane micro-optimization, anyway).

**Comment by tseaver** [comment]
@hannosch Maybe ready for a new `DocumentTemplate` release?  I can confirm that the ZMI frames all render apparently-correctly with today's two changes on Python 3.6.

**Comment by hannosch** [comment]
@tseaver Feel free to cut a new release ;)

**Comment by tseaver** [comment]
:p https://pypi.python.org/pypi/DocumentTemplate/3.0a4

**Pull request merged**